### PR TITLE
feat(frontend): Redesign of the Flow Editor + Arbitrary forloop

### PR DESCRIPTION
**Pull request by fatonramadani**
This PR contains the code to support for loops as a first-class step in the frontend. It aims to match all the features already available in the backend.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
 		"lint": "prettier --ignore-path .gitignore --check --plugin-search-dir=. . && eslint --ignore-path .gitignore .",
 		"format": "prettier --ignore-path .gitignore --write --plugin-search-dir=. .",
 		"package": "svelte-kit package && cd package && rm README.md && rm README_DEV.md && sed -i -e 's/windmill/windmill-components/g' package.json",
-		"generate-backend-client": "openapi --input ../backend/openapi.yaml --output ./src/lib/gen --useOptions && sed -i '213 i \\    request.referrerPolicy = \"no-referrer\"\n' src/lib/gen/core/request.ts",
+		"generate-backend-client": "openapi --input ../backend/openapi.yaml --output ./src/lib/gen --useOptions",
```
Comment (fatonramadani) [inline frontend/package.json:13]: @rubenfiszel It looks like this cmd is outdated, can you please confirm ?

Comment (rubenfiszel) [inline frontend/package.json]: please revert, this is so we do not send the referrer containing the whole state at every request of thr script or flow editor

File: frontend/src/lib/components/FlowBuilder.svelte [hunk 2]
```diff
@@ -125,10 +127,12 @@
 		goto(`?step=${step}`)
 	}
 
+	flowStateStore.subscribe((flowState: FlowState) => {
+		flowStore.update((flow: Flow) => flowStateToFlow(flowState, flow))
```
Comment (rubenfiszel) [inline frontend/src/lib/components/FlowBuilder.svelte]: gotta check if flowState is not undefined I think. flowState &&

File: frontend/src/lib/components/FlowBuilder.svelte [hunk 3]
```diff
@@ -125,10 +127,12 @@
...
+	})
+
 	flowStore.subscribe((flow: Flow) => {
-		if (flow) {
-			setQueryWithoutLoad($page.url, 'state', encodeState(flowToMode(flow, $mode)))
-		}
+		setQueryWithoutLoad($page.url, 'state', encodeState(flow))
```
Comment (rubenfiszel) [inline frontend/src/lib/components/FlowBuilder.svelte]: gotta check if flow is undefined

**Pull request merged**